src/databricks/labs/ucx/source_code/files.py

```diff
@@ -100,7 +213,16 @@ def load_dependency(self, path_lookup: PathLookup, dependency: Dependency) -> So
         absolute_path = path_lookup.resolve(dependency.path)
         if not absolute_path:
             return None
-        return LocalFile(absolute_path, absolute_path.read_text("utf-8"), Language.PYTHON)
+        # for now we only support python
+        if not absolute_path.as_posix().endswith(".py"):
+            return None
+        for encoding in ("utf-8", "ascii"):
+            try:
+                code = absolute_path.read_text(encoding)
```

Comment:
read magic header and return a notebook container in case of a match.

Reply:
that's done by the FileLinter